pkg/ccl/streamingccl/streamproducer/producer_job.go

```diff
@@ -84,12 +100,22 @@ func (p *producerJobResumer) Resume(ctx context.Context, execCtx interface{}) er
 			if err != nil {
 				return err
 			}
-			// The job completes successfully if the ingestion has been cut over.
-			if p := j.Progress(); p.GetStreamReplication().IngestionCutOver {
-				return nil
-			}
-			if isTimedOut(j) {
-				return errors.Errorf("replication stream %d timed out", p.job.ID())
+
+			prog := j.Progress()
+			switch prog.GetStreamReplication().StreamIngestionStatus {
+			case jobspb.StreamReplicationProgress_FINISHED_SUCCESSFULLY:
+				return p.releaseProtectedTimestamp(ctx, execCfg)
+			case jobspb.StreamReplicationProgress_FINISHED_UNSUCCESSFULLY:
+				return j.Update(ctx, nil, func(txn *kv.Txn, md jobs.JobMetadata, ju *jobs.JobUpdater) error {
+					ju.UpdateStatus(jobs.StatusCancelRequested)
```

Comment:
Fine for now, but lets think through whether we want to release the protected timestamp in the case of it finishing unnecessfully.